### PR TITLE
Add firm index for self-service dashboard

### PR DESCRIPTION
**Pull request by neoeno**
NB: this branch contains a commit marked `[KILL]` that will be removed and replaced by a version of `mas-rad_core` when https://github.com/moneyadviceservice/mas-rad_core/pull/60 is merged.

### CODE REVIEW

File: app/assets/stylesheets/components/firm/_all.scss [hunk 1]
```diff
@@ -1,2 +1,3 @@
 @import 'settings';
 @import 'firm';
+@import 'firm_list';
```
Comment (benbarnett) [inline app/assets/stylesheets/components/firm/_all.scss]: Really `firm_list` should be moved into its own directory within components, and have its own settings file.

File: app/assets/stylesheets/components/firm/_firm_list.scss [hunk 2]
```diff
@@ -0,0 +1,27 @@
+$color-firm-list-border-grey: #ccc;
```
Comment (benbarnett) [inline app/assets/stylesheets/components/firm/_firm_list.scss]: This colour can then be moved into settings file mentioned in previous comment

File: app/assets/stylesheets/components/_navigation.scss [hunk 3]
```diff
@@ -1,10 +1,10 @@
-.rad__authentication {
+.rad__navigation {
```
Comment (benbarnett) [inline app/assets/stylesheets/components/_navigation.scss]: I know this is just a rename but what do you think about losing the `rad__` prefix? I'm unsure what purpose it serves as there doesn't appear to be a `rad` block that it's inheriting from? If it was due to a naming clash then `rad-navigation` would work better.

File: app/views/shared/_navigation.html.erb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<nav class="t-navigation">
+  <ul class="rad__navigation">
+    <% unless user_signed_in? %>
+      <li class='rad__navigation__item'>
+        <%= link_to t('authentication_sign_in'), new_user_session_path, class: 'rad__navigation__link' %>
```
Comment (benbarnett) [inline app/views/shared/_navigation.html.erb]: (If you agree with removing `rad__` then it'll affect these lines too.)

File: app/assets/stylesheets/components/firm/_all.scss (hunk 1 above)
Comment (neoeno) [inline app/assets/stylesheets/components/firm/_all.scss]: Fair point — didn't think about the directory structure beyond 'this is to do with firms'. But it's about the first BEM component right? Will action.

Comment (benbarnett) [inline app/assets/stylesheets/components/firm/_all.scss]: Yeah exactly, the directory contains the component & any settings. So unless `firm_list` was a child of `firm` (which it isn't) then it deserves its own directory.

File: app/assets/stylesheets/components/_navigation.scss (hunk 3 above)
Comment (neoeno) [inline app/assets/stylesheets/components/_navigation.scss]: Ah yes, good point. It was initially due to a naming clash with frontend's `.authentication`, but now we've renamed it we don't have to worry about that anymore.

**Comment by neoeno** [comment]
@benbarnett all actioned

**Comment by benbarnett** [comment]
Nice one. :+1: it is!

**Comment by JiggyPete** [comment]
:+1:

**Pull request merged**